controller/app/pending_ops_models/remove_alias_op.rb

```diff
@@ -31,6 +33,8 @@ def rollback
       application.save!
     end
 
+    OpenShift::SsoService.register_gear(gear) unless gear.removed
```

Comment:
Register_gear always defines the current state of the gear/application so that it can be used for any operation. The information needed from the gear/application is the FQDN(s) that can be used to access an application, including any aliases, and the contents of some environment variables that will be used to define the associated configuration.